src/components/RewardBudget/RewardBudget.tsx

```diff
@@ -70,20 +72,27 @@ export const RewardBudget: FC<RewardBudgetProps> = props => {
     <div className="flex w-full flex-col rounded-2xl border border-greyscale-50/8 bg-whiteFakeOpacity-2 shadow-light">
       <div className="flex w-full flex-col px-8 pb-2 pt-4">
         <div className="flex w-full flex-col gap-2">
-          <div className="z-1 flex flex-1 flex-row items-stretch justify-between gap-4">
+          <div className="z-10 flex flex-1 flex-row items-stretch justify-between gap-4">
             <RewardBudgetSelect {...props} value={selectedBudget} onChange={onSelectedBudgetChange} />
             <FieldInput
               min="0"
+              step="0.01"
+              placeholder={T(`currencies.amount_placeholder.${selectedBudget.currency}`)}
               name="budget-amount-input"
               type="number"
-              value={amount}
-              fieldClassName="flex-1 max-w-[60px] min-w-[60px] w-[60px]"
-              className="h-full flex-1"
+              value={amount === undefined ? "" : amount}
```

Comment:
Prefer truthy ternaries
```suggestion
              value={Number.isFinite(amount) ? amount : ""}
```